docs/auth.md

```diff
@@ -0,0 +1,66 @@
+---
+id: auth
+title: "Authorization with OAuth2"
+sidebar_label: "OAuth2"
+---
+
+**ZIO CLI** can perform interactions requiring OAuth2. OAuth2 is added to a CLI App as an `Options[OAuth2Token]`. The token can be stored and the user can specify the path, so it is not necessary to repeat authentication. We can create it as with other options:
+```scala mdoc:silent
+import zio.cli._
+import zio.cli.oauth2.OAuth2Provider
+
+val clientId = "clientId"
+
+val provider: OAuth2Provider = OAuth2Provider.Github(clientId)
+val scope: List[String] = List("repo")
+val oauth2 = Options.oauth2(provider, scope)
+```
+
+## Construction
+Currently the supported OAuth2 providers are GitHub, Google and Facebook.
+```scala mdoc:silent
+
+val clientSecret = "clientSecret"
+val appId = "appId"
+val clientToken = "clientToken"
+
+val githubOAuth = Options.oauth2(OAuth2Provider.Github(clientId), List("repo"))
+
+val googleOAuth = Options.oauth2(OAuth2Provider.Google(clientId, clientSecret), Nil)
+
+val facebookOAuth = Options.oauth2(OAuth2Provider.Facebook(appId, clientToken), Nil)
+
+```
+
+## Full example
+This example shows how to integrate OAuth2 in a ZIO `CliApp`. We are going to make a CLI App that interacts with Github and has two commands:
```

Comment:
Rather than just 2 sentences of text, followed by code, it would be good to walk the user, step-by-step, through building a tiny little CLI that supports OAuth2 authentication.